### PR TITLE
Improving sphinx docs

### PR DESCRIPTION
**Pull request by macartur**

### CODE REVIEW

**Comment by beraldoleal** [comment]
@macartur Please create another PR with the same name and content also to branch 2017.1rc

File: setup.cfg [hunk 1]
```diff
@@ -4,7 +4,7 @@ exclude = .eggs,ENV,build,docs/conf.py,venv
 [pylama]
 linters = isort,pydocstyle,radon,pylint,pycodestyle,pyflakes
 skip = */v0x02/*
-ignore = D105,D203,D213,I0011
+ignore = D105,D203,D213,I0011,D301
```
Comment (cemsbr) [inline setup.cfg]: I noticed you are using r""", so why ignoring D301?

File: pyof/foundation/base.py [hunk 2]
```diff
@@ -340,14 +340,14 @@ def _header_message_type_update(obj, attr):
 
     @staticmethod
     def get_pyof_version(module_fullname):
-        """Get the module pyof version based on the module fullname.
+        r"""Get the module pyof version based on the module fullname.
```
Comment (diraol) [inline pyof/foundation/base.py]: @macartur by keeping this `r"""` your `\` from line 350 will not work, right?

File: pyof/foundation/base.py [hunk 3]
```diff
@@ -386,7 +386,7 @@ def replace_pyof_version(module_fullname, version):
 
     @staticmethod
     def get_pyof_obj_new_version(name, obj, new_version):
-        """Return a class atrribute on a different pyof version.
+        r"""Return a class attribute on a different pyof version.
```
Comment (diraol) [inline pyof/foundation/base.py]: Same above comment applies here.

File: pyof/v0x01/common/action.py [hunk 4]
```diff
@@ -56,9 +57,9 @@ class ActionHeader(GenericStruct):
 
     action_type = UBInt16(enum_ref=ActionType)
     length = UBInt16()
-    #: Pad for 64-bit alignment.
-    #: This attribute will not be implemented since not all subclasses from
-    #: this class will hold it on the same place and with the same size.
+    # Pad for 64-bit alignment.
```
Comment (diraol) [inline pyof/v0x01/common/action.py]: @cemsbr should we use the `#:` or the `# ` over here?

File: pyof/foundation/base.py (hunk 3 above)
Comment (macartur) [inline pyof/foundation/base.py]: @diraol, in this case, is necessary. Below we have a binary number b'\x01', and the symbol '\' is used.

File: pyof/v0x01/common/action.py (hunk 4 above)
Comment (cemsbr) [inline pyof/v0x01/common/action.py]: This is OK because the attribute in line 63 is commented out.

**Pull request merged**